options/ansi/generic/time-stubs.cpp

```diff
@@ -578,10 +579,445 @@ char *ctime_r(const time_t *clock, char *buf) {
 	return asctime_r(localtime(clock), buf);
 }
 
-char *strptime(const char *__restrict, const char *__restrict,
+namespace {
+
+int read_number(int *result, size_t *string_index, const char *s, int ndigits) {
+    int product = 0;
+    int count = 0;
+    int si = *string_index;
+    for(size_t j = 0; j < ndigits; j++)
+        if(isdigit(s[si])){
+            product = product * 10 + (s[si++] - '0');
+            count++;
+        } else break;
+    *result = product;
+    *string_index = si;
+    return count;
+}
+
+int month_to_day(int month) {
+    switch(month){
+        case  0: return 0;
+        case  1: return 31;
+        case  2: return 59;
+        case  3: return 90;
+        case  4: return 120;
+        case  5: return 151;
+        case  6: return 181;
+        case  7: return 212;
+        case  8: return 243;
+        case  9: return 273;
+        case 10: return 304;
+        case 11: return 334;
+    }
+    return -1;
+}
+
+int is_leapyear(int year) {
+    return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
+}
+
+int month_and_year_to_day_in_year(int month, int year){
+    int day = month_to_day(month);
+    if(is_leapyear(year) && day < month_to_day(2)) //TODO just do 2 instead of month_to_day(2) lol
```

Comment:
Comments should be formal

Reply:
This was left as a todo for myself that I forgot about.